packages/integrations/source-control/src/parse-hunks.ts

```diff
@@ -0,0 +1,172 @@
+export type Change = UnknownChange | InsertChange | DeleteChange | NormalChange;
+
+export type UnknownChange = {
+  content: string;
+  type: 'unknown';
+  oldLineNumber: number;
+  newLineNumber: number;
+}
+
+export type InsertChange = {
+  content: string;
+  type: 'insert';
+  lineNumber: number;
+}
+
+export type DeleteChange = {
+  content: string;
+  type: 'delete';
+  lineNumber: number;
+}
+
+export type NormalChange = {
+  content: string;
+  type: 'normal';
+  oldLineNumber: number;
+  newLineNumber: number;
+}
+
+export type Hunk = {
+  content: string;
+  oldStart: number;
+  newStart: number;
+  oldLines: number;
+  newLines: number;
+  additions: number;
+  deletions: number;
+  internals: DIC;
+  changes: Change[];
+}
+
+const firstLineRegex = /^@@\s+-([0-9]+)(,([0-9]+))?\s+\+([0-9]+)(,([0-9]+))?/;
+const ADCRegex = /n*d*i*/g;
+
+const countLetters = (str: string) => {
+  const array = [...str];
+  const d = array.filter(c => c === 'd').length;
+  const i = array.filter(c => c === 'i').length;
+
+  const c = Math.min(d, i);
+
+  return {
+    d: d - c,
+    i: i - c,
+    c,
+  }
+}
```

Comment:
pls us lngr nms fr thngs

If d = deletions
And i = insertions
Why is c(hanges?) min of both ?

Reply:
no.

Reply:
These are for future use -> I have just implemented them right now.

These are for analyzing context of an PR -> We will need to find out how long it takes to review a PR.

c -> are number of lines that are deleted and new one inserted in their place. -> my educated guess is that these line are the hardest to read.

That's why dic.